src/Processors/Transforms/SquashingChunksTransform.cpp

```diff
@@ -64,8 +69,22 @@ void SimpleSquashingChunksTransform::transform(Chunk & chunk)
     }
     else
     {
+        if (chunk.hasRows())
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "Chunk expected to be empty, otherwise it will be lost");
+
         auto block = squashing.add({});
         chunk.setColumns(block.getColumns(), block.rows());
+
+        /// ISimpleTransform keeps output chunk (result of transform() execution) for some time and push it in the output port within subsequent prepare() call.
+        /// Because of our custom prepare() implementation we have to take care of both places where data could be buffered: `output_data` and `squashing`.
+        if (output_data.chunk.hasRows())
+        {
+            auto res = std::move(output_data.chunk);
+            output_data.chunk.clear();
```

Comment:
Isn't access to moved object undefined behavior?

Reply:
I checked this PR because of `SimpleSquashingChunksTransform`, but I found this comment and I couldn't stand not to answer (because this is my all time favourite answer in terms of C++): the "moved-from-objects" are in **a valid, but unspecified state**. That means reading from them might be tricky, but calling any function on them that doesn't have any preconditions is fine. In other worlds you can use the objects as you want, you just have to "forget" everything you know about them. An [SO answer](https://stackoverflow.com/a/7028318/6639989) explaining this with a bit more context